### PR TITLE
[nrf fromtree] trusted-firmware-m: Configurable SECURE_UART1

### PR DESCRIPTION
**Pull request by Vge0rge**
-This disables the configuration of the
 SECURE_UART1 in preload. The intention is to make it
 configurable in a Kconfig option.

Upstream PR: [11569](https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/11569)

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

### CODE REVIEW

**Review by oyvindronningstad — APPROVED** [review]

File: trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf5340/preload.cmake [hunk 1]
```diff
@@ -14,5 +14,4 @@
 # Set architecture and CPU
 set(TFM_SYSTEM_PROCESSOR cortex-m33)
 set(TFM_SYSTEM_ARCHITECTURE armv8-m.main)
-set(TFM_SYSTEM_DSP OFF)
-set(SECURE_UART1 ON)
+set(TFM_SYSTEM_DSP OFF)
```
Comment (tejlmand) [inline trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf5340/preload.cmake:17]: please fix newline.

File: trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf9160/preload.cmake [hunk 2]
```diff
@@ -14,5 +14,4 @@
 # Set architecture and CPU
 set(TFM_SYSTEM_PROCESSOR cortex-m33)
 set(TFM_SYSTEM_ARCHITECTURE armv8-m.main)
-set(TFM_SYSTEM_DSP OFF)
-set(SECURE_UART1 ON)
+set(TFM_SYSTEM_DSP OFF)
```
Comment (tejlmand) [inline trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf9160/preload.cmake:17]: please fix newline.

**Review by tejlmand — CHANGES_REQUESTED** [review]
please fix newline.

File: trusted-firmware-m/platform/ext/common/tfm_platform.c [hunk 3]
```diff
@@ -13,7 +13,11 @@
 __WEAK enum tfm_hal_status_t tfm_hal_platform_init(void)
 {
     __enable_irq();
+
+    /* Only if UART1 is used by TF-M do we initialize it. */
+#ifdef SECURE_UART1
```
Comment (tejlmand) [inline trusted-firmware-m/platform/ext/common/tfm_platform.c:18]: How will this impact boards that are having `SECURE_UART1` to off ?

For example the Zephyr board `mps2_an521_ns` which is frequently been used for testing.

File: trusted-firmware-m/platform/ext/common/tfm_platform.c [hunk 4]
```diff
@@ -13,7 +13,11 @@
 __WEAK enum tfm_hal_status_t tfm_hal_platform_init(void)
 {
     __enable_irq();
+
```
Comment (tejlmand) [inline trusted-firmware-m/platform/ext/common/tfm_platform.c:16]: Nothing wrong in this line, but please refer to the upstream PR in the commit message.

And note, this fork is a fork from here:
https://github.com/zephyrproject-rtos/trusted-firmware-m/

not here:
https://review.trustedfirmware.org/plugins/gitiles/TF-M/trusted-firmware-m/+/refs/heads/master

so please refer to the correct upstream PR to make life easier during upmerges.

**Review by tejlmand — CHANGES_REQUESTED** [review]
please refer to upstream PR in the `[nrf fromlist]` commit message.

File: trusted-firmware-m/platform/ext/common/tfm_platform.c (hunk 3 above)
Comment (Vge0rge) [reply trusted-firmware-m/platform/ext/common/tfm_platform.c:18]: I am aware of any methods to test the code with external boards, but let me know if you are aware of any.

File: trusted-firmware-m/platform/ext/common/tfm_platform.c (hunk 4 above)
Comment (Vge0rge) [reply trusted-firmware-m/platform/ext/common/tfm_platform.c:16]: I am a bit confused, what do you mean the correct PR? 
Should I open a new PR here:
https://github.com/zephyrproject-rtos/trusted-firmware-m/

Never did that until now I think.

Comment (tejlmand) [reply trusted-firmware-m/platform/ext/common/tfm_platform.c:16]: nevermind, seems like in our tf-m fork the most common thing is to refer to the up-upstream repo, and not just to the first upstream repo (Zephyr's fork)

**Review by tejlmand — APPROVED** [review]

**Review by tejlmand — CHANGES_REQUESTED** [review]
sorry, newlines still not fixed.

File: trusted-firmware-m/platform/ext/common/tfm_platform.c (hunk 3 above)
Comment (Vge0rge) [reply trusted-firmware-m/platform/ext/common/tfm_platform.c:18]: The current solution does not impact any other boards.

File: trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf5340/preload.cmake (hunk 1 above)
Comment (Vge0rge) [reply trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf5340/preload.cmake]: Added new line so this should be fixed.

File: trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf9160/preload.cmake (hunk 2 above)
Comment (Vge0rge) [reply trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf9160/preload.cmake]: Added new line so this should be fixed.

**Review by hakonfam — CHANGES_REQUESTED** [review]
Please refer to the upstream PR in your commit message. Also note that https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/11569 is merged and should be a fromtree :)

**Comment by Vge0rge** [comment]
> Please refer to the upstream PR in your commit message. Also note that https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/11569 is merged and should be a fromtree :)

Done

File: trusted-firmware-m/platform/ext/common/tfm_platform.c [hunk 5]
```diff
@@ -13,7 +13,11 @@
...
+    /* Only if UART1 is used by TF-M do we initialize it. */
+#ifdef SECURE_UART1
     stdio_init();
+#endif
```
Comment (SebastianBoe) [inline trusted-firmware-m/platform/ext/common/tfm_platform.c:20]: I thought we weren't going to use the __WEAK implementation anymore, so no need to patch it?

Comment (Vge0rge) [reply trusted-firmware-m/platform/ext/common/tfm_platform.c]: True, mixed up this file. Now only the removal of the lines in the preload file is there.

**Review by SebastianBoe — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Review by hakonfam — APPROVED** [review]

**Pull request merged**